rep-2004.rst

```diff
@@ -434,86 +354,117 @@ For example, this might be a tutorial package which is not intended for reuse bu
+   #. No explicit testing requirements, though covering some if not all of the features with tests is recommended
 
-* Dependencies:
+#. **Dependencies:**
 
-  * No restrictions
+   #. No restrictions
 
-* Platform Support:
+#. **Platform Support:**
 
-  * May support all tier 1 platforms for ROS 2, as defined in `REP-2000 <https://www.ros.org/reps/rep-2000.html#support-tiers>`_
+   #. May support all tier 1 platforms for ROS 2, as defined in `REP-2000 <https://www.ros.org/reps/rep-2000.html#support-tiers>`_
 
 Any package that does not claim to be 'Level 3' or higher is automatically 'Level 4'.
-Refer to the detailed description of the requirements in the Quality Level 1 section above for more information.
+Refer to the detailed description of the requirements following the Quality Level 1 section above for more information.
 
 Quality Level 5
 ^^^^^^^^^^^^^^^
 
-Packages in this category simply do not meet even the 'Level 4' requirements, and for that reason should not be used.
-The rationale being that all packages should have at least a declare license or licenses and should include copyright statements in each file.
+Packages in this category cannot even meet the simple 'Level 4' requirements, and for that reason should not be used.
+All packages should have at least a declared license or set of licenses and should include copyright statements in each file.
 
 Repository Organization
 ^^^^^^^^^^^^^^^^^^^^^^^
 
 Since these categories are applied on a per package basis, and since there may be more than one package per source repository, it's recommended that the strictest set of policies apply to the whole repository.
-This is recommended, rather than trying to mix processes depending on which packages are changed in a given change request (pull request or merge request, etc.).
+This is recommended rather than trying to mix processes depending on which packages are changed in a given change request (pull request or merge request, etc.).
 If this is too onerous, then it's recommended to split lower quality packages out into a separate repository.
 
+Reference Implementation
+========================
 
-Motivation
-==========
+The `ROS 2 Developer Guide <https://index.ros.org/doc/ros2/Contributing/Developer-Guide/>`_ describes the policies we implement to achieve Quality Level 1 for ROS Core packages.
 
-TBD
+The `rcutils package's quality declaration <https://github.com/ros2/rcutils/pull/202/files>`_ is one example of the conditions of this REP in practice on a non-trivial package.
 
+.. update link when that draft is merged
 
-Rationale
-=========
+Quality Declaration Template
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-TBD
+.. code-block:: markdown
 
+  This document is a declaration of software quality for the `<package name>` package, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).
 
-Backwards Compatibility
-=======================
+  # `<package name>` Quality Declaration
 
```

Comment:
@wjwwood rep0001 says the Backwards Compatibility section doesn't need to be included if there isn't anything to say - just double checking